From e66fbc3f12970cb582964ada8156612bc97fd4f9 Mon Sep 17 00:00:00 2001
From: Ma Wupeng <mawupeng1@huawei.com>
Date: Mon, 27 Mar 2023 19:22:31 +0800
Subject: [PATCH 1625/1646] [Huawei] mm/dpool: mem_reliable: Stop alloc
 reliable page from dynamic pool

Offering: HULK
hulk inclusion
category: feature
bugzilla: 188703

--------------------------------

With memory reliable enabled, user expect to alloc memory from mirrored
regions (also known as the non-movable zone). However dynamic pool does
not contain such reliable information. So bypass it.

Signed-off-by: Ma Wupeng <mawupeng1@huawei.com>
---
 mm/dynamic_hugetlb.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/mm/dynamic_hugetlb.c b/mm/dynamic_hugetlb.c
index 85cdcd74b27a..8cd574d73437 100644
--- a/mm/dynamic_hugetlb.c
+++ b/mm/dynamic_hugetlb.c
@@ -511,6 +511,11 @@ static bool should_allocate_from_dhugetlb_pool(gfp_t gfp_mask)
 	if ((gfp | __GFP_IO | __GFP_FS) != GFP_HIGHUSER_MOVABLE)
 		return false;
 
+#ifdef CONFIG_MEMORY_RELIABLE
+	if (mem_reliable_is_enabled() && (gfp_mask & GFP_RELIABLE))
+		return false;
+#endif
+
 	return true;
 }
 
-- 
2.17.1

